package.json

```
@@ -7,10 +7,17 @@
     "url": "git://github.com/kysely-org/kysely.git"
   },
   "engines": {
-    "node": ">=14.0.0"
+    "node": ">=18.0.0"
   },
   "main": "dist/cjs/index.js",
   "module": "dist/esm/index.js",
+  "typesVersions": {
+    "<4.6": {
+      "*": [
+        "outdated-typescript.d.ts"
+      ]
+    }
+  },
   "exports": {
     ".": {
       "import": "./dist/esm/index.js",
```

According to the TS [documentation](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html#exports-is-prioritized-over-typesversions), the `exports` key takes precedence over the `typesVersions` key.

This change was technically made in TS 4.9, but one of the maintainers indicated [here](https://github.com/microsoft/TypeScript/pull/50890#issuecomment-1256406529) that non-`exports` keys cannot contribute to the resolution when an `exports` key is present, so this should still be applicable.

```suggestion
  "exports": {
    ".": {
      "types@<4.6": "outdated-typescript.d.ts",
      "types": "./dist/esm/index.d.ts",
      "import": "./dist/esm/index.js",
```

---

Hey 👋 

You're commenting on the wrong PR.

Anyway, this still works for the version being targeted. If the cutoff eventually happens on a version where `typesVersions` is no good anymore, we'll change that.